src/rust/engine/src/nodes.rs

```diff
@@ -336,31 +334,21 @@ impl MultiPlatformExecuteProcess {
       .map(|(name, dest)| Ok((CacheName::new(name)?, CacheDest::new(dest)?)))
       .collect::<Result<_, String>>()?;
 
-    let jdk_home = {
-      let val = externs::getattr_as_string(value, "jdk_home");
-      if val.is_empty() {
-        None
-      } else {
-        Some(PathBuf::from(val))
-      }
-    };
+    let jdk_home = externs::getattr_as_optional_string(py, value, "jdk_home").map(PathBuf::from);
 
     let py_use_nailgun: Value = externs::getattr(value, "use_nailgun").unwrap();
     let use_nailgun = lift_directory_digest(&py_use_nailgun)
       .map_err(|err| format!("Error parsing use_nailgun {}", err))?;
 
-    let execution_slot_variable = {
-      let s = externs::getattr_as_string(value, "execution_slot_variable");
-      if s.is_empty() {
-        None
-      } else {
-        Some(s)
-      }
-    };
+    let execution_slot_variable =
+      externs::getattr_as_optional_string(py, value, "execution_slot_variable");
 
-    let cache_scope =
-      externs::getattr_as_string(&externs::getattr(value, "cache_scope").unwrap(), "name")
-        .try_into()?;
+    let cache_scope: ProcessCacheScope = {
+      let cache_scope_enum: PyObject = externs::getattr(value, "cache_scope").unwrap();
```

Comment:
Maybe avoid the unwrap? `.ok_or_else(|| "Missing cache_scope attribute".into())?`

Reply:
Similar comment applies to other locations with `.unwrap`

Reply:
I think that might be even more confusing. Really, this could should not be calling `.unwrap()` and should propagate the error. But error handling is clunky in rust-cpython (much better with PyO3). My main focus is getting PyO3 working with making as few changes as feasible. Then, once PyO3 lands, we can start improving the code like this.